components/console/introduction.rst

```diff
@@ -110,6 +110,31 @@ This prints::
 
     HELLO FABIEN
 
+Command Lifecycle
+~~~~~~~~~~~~~~~~~
+
+Commands have three lifecycle methods:
+
+:method:`Symfony\\Component\\Console\\Command\\Command::initialize`
+
+    This method is executed before the ``interact()`` and the ``execute()``
+    methods. It's main purpose is to initialize the variables used in the
+    rest of the command methods.
+
+:method:`Symfony\\Component\\Console\\Command\\Command::interact`
+
+    This method is executed after ``initialize()`` and before ``execute()``.
+    Its purpose is to check if some of the options/arguments are missing
+    and interactively ask the user for those values.
```

Comment:
we should add that this method is executed *before* the input definition is validated. This means that this is the *only* place where you can ask for missing options, otherwise the command throw an error about missing options.